pkg/oauthflow/device.go

```diff
@@ -0,0 +1,157 @@
+/*
+Copyright © 2021 The Sigstore Authors
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package oauthflow
+
+import (
+	"context"
+	"encoding/json"
+	"errors"
+	"fmt"
+	"io/ioutil"
+	"net/http"
+	"net/url"
+	"time"
+
+	"github.com/coreos/go-oidc/v3/oidc"
+	"golang.org/x/oauth2"
+)
+
+const (
+	/* #nosec */
+	SigstoreDeviceURL = "https://oauth2.sigstore.dev/auth/device/code"
+	/* #nosec */
+	SigstoreTokenURL = "https://oauth2.sigstore.dev/auth/device/token"
+)
+
+type deviceResp struct {
```

Comment:
might be useful to get the `expires_in` timestamp and print it out to the console

Reply:
Nice!